### PR TITLE
Automatically generate a unique name for registries

### PR DESCRIPTION
**Pull request by nturinski**
Because registry names have to be globally unique, it's almost guaranteed to have failed the workspace name check. This ensures that the registry name will almost always be unique.

### CODE REVIEW

File: src/commands/image/imageSource/containerRegistry/acr/createAcr/RegistryNameStep.ts [hunk 1]
```diff
@@ -51,4 +52,29 @@ export class RegistryNameStep extends AzureWizardPromptStep<CreateAcrContext> {
             return { nameAvailable: true };
         }
     }
+
+    public static async tryGenerateRelatedName(context: DeployWorkspaceProjectContext, name: string): Promise<string | undefined> {
+        let registryAvailable: boolean = false;
+        let generatedName: string = '';
+
+        const timeoutSeconds: number = 60;
```
Comment (MicroFish91) [inline src/commands/image/imageSource/containerRegistry/acr/createAcr/RegistryNameStep.ts:60]: Although rare that we would ever hit the full length of this timeout, I'm wondering if the timeout is too long to be hanging on an input name check.  Not really sure to be honest but just thought I'd bring it up, let me know what you think

**Review by MicroFish91 — APPROVED** [review]
Everything looks good! :D 

I'll run a few extra tests tonight just to double check though, will follow-up when finished with a comment

File: src/commands/image/imageSource/containerRegistry/acr/createAcr/RegistryNameStep.ts (hunk 1 above)
Comment (nturinski) [reply src/commands/image/imageSource/containerRegistry/acr/createAcr/RegistryNameStep.ts:60]: Yeah, I could see that. Maybe just try for 10 seconds?

Comment (MicroFish91) [reply src/commands/image/imageSource/containerRegistry/acr/createAcr/RegistryNameStep.ts:60]: Yeah I personally think I'd be happy with anything between 10-30s.

**Comment by MicroFish91** [comment]
I tested this with quite a few scenarios and all of them seem to work great, so probably good to merge :)

**Pull request merged**